cip/CIP2015-06-24-call-procedures.adoc

```diff
@@ -0,0 +1,250 @@
+
+As a basic example, consider a call to the stored procedure `myProc(name``::``STRING?``,``id``::``INTEGER?)``::`` (last``::``STRING?)`:
+
+.Calling stored procedure inside a larger query
+[source, cypher]
+----
+MATCH (n:Person)-[r:IN]->(g:Group)
+CALL myProc(n.name, g.id*1000+r.id) YIELD last AS lastLogin
+RETURN *
+----
+
+This calls `myProc` for each record produced by the preceding `MATCH` clause with a `name` argument obtained by evaluating `n.name` and an `id` argument obtained by evaluating `g.id*1000+r.id`.
+Each call may yield multiple procedure result records with a single result field `last`.
+For each of them, a new statement result record is created that contains the original variables already in scope (i.e. `n`, `r`, `g`) as well as the projected result field `last` renamed as `lastLogin`.
+
+`YIELD -` is used to indicate that no result fields should be projected.
+In this case the procedure is still called; execution of the call just does not introduce any new variable bindings.
+
+However, the `YIELD` subclause is *always* omitted if the procedure returns nothing (i.e. is of type `VOID`).
+As an example, consider a call to the stored procedure `myVoidProc(name``::``STRING?``,``id``::``INTEGER?)``::``VOID`:
+
+.Calling a stored procedure that returns nothing
+[source, cypher]
+----
+MATCH (n:Person)-[r:IN]->(g:Group)
+CALL myVoidProc(n.name, g.id*1000+r.id)
+RETURN n.name AS name
+----
+
+This will behave in the same way as above but since the procedure now returns nothing, calling it will not change the result cardinality, i.e. conceptually the procedure is called once for each record produced by `MATCH` and will just pass that record on to `RETURN`.
+
+==== Standalone calls
+
+Procedures may also be called standalone, i.e. without taking arguments from or combining their results with other parts of a larger query.
+In this case, the trailing `RETURN` clause may be omitted and all variable in scope are returned implicitly by the statement.
+Furthermore, we propose two optional simplifications that are available for standalone calls only.
+These simplifications are outlined in the following:
+
+===== Calling with implicitly passed arguments (parameters)
+
+Standalone calls may ommit passing arguments explicitly.  In this case, all required procedure arguments are taken implicitly from statement parameters with the same name.
+
+Again consider a call to the stored procedure `myProc(name``::``STRING?``,``id``::``INTEGER?)``::`` (last``::``STRING?)`:
+
+.Standalone call to a stored procedure using implicitly passed arguments
+[source, cypher]
+----
+CALL myProc YIELD last AS lastLogin
+----
+
+This is the same as executing:
+
+.Standalone call to a stored procedure using explicitly passed arguments
+[source, cypher]
+----
+CALL myProc({name}, {id}) YIELD last AS lastLogin
```

Comment:
Hmm... this is tricky... ought '{name}' etc be changed to '$name' to reflect our new param syntax?